src/psyclone/tests/psyir/backend/fortran_test.py

```diff
@@ -793,6 +795,7 @@ def test_fw_container_2(fort_writer):
         "  real :: c\n"
         "  real :: d\n\n"
         "  public :: tmp\n\n"
+        "  implicit none\n"
```

Comment:
I'm afraid this won't compile as the `implicit none` must come before any variable declarations (but after any `use` statements). Is it possible to add compilation to this test (and others you've touched)?

Reply:
Fixed implicit none placement and added compilation check